scripts/train_agents.py

```diff
@@ -453,40 +458,67 @@ def N_X_SP(args,
-        n_x_sp_train_types = curriculum.train_types,
-        n_x_sp_eval_types=args.primary_eval_types,
-        n_x_sp_total_training_timesteps=args.n_x_sp_total_training_timesteps,
-        n_x_sp_force_training=primary_force_training,
+        n_x_sp_train_types=curriculum.train_types,
+        n_x_sp_eval_types=primary_eval_types,
         curriculum=curriculum,
-        parallel=parallel,
-        num_SPs_to_train=args.num_SPs_to_train
+        unseen_teammates_len=unseen_teammates_len,
         )
 
 
-def N_1_SP(args, 
-            pop_force_training:bool,
-            primary_force_training:bool,
-            parallel:bool) -> None:
+def N_X_SP_w_adversaries(args) -> None:
+    attack_rounds = 3
+    unseen_teammates_len = 1
+    adversary_play_config = AdversaryPlayConfig.MAP
+    primary_train_types = [TeamType.SELF_PLAY_HIGH, TeamType.SELF_PLAY_HIGH, TeamType.SELF_PLAY_HIGH, TeamType.SELF_PLAY_HIGH,
+                           TeamType.SELF_PLAY_MEDIUM, TeamType.SELF_PLAY_MEDIUM, TeamType.SELF_PLAY_MEDIUM, TeamType.SELF_PLAY_MEDIUM,
+                           TeamType.SELF_PLAY_ADVERSARY]
+
+    primary_eval_types = {'generate': [TeamType.SELF_PLAY_HIGH, TeamType.SELF_PLAY_LOW, TeamType.SELF_PLAY_ADVERSARY],
+                          'load': []}
+
+    curriculum = Curriculum(train_types = primary_train_types,
+                            is_random=False,
+                            total_steps = args.n_x_sp_total_training_timesteps//args.epoch_timesteps,
+                            training_phases_durations_in_order={
+                                TeamType.SELF_PLAY_ADVERSARY: 0.5,
+                            },
+                            rest_of_the_training_probabilities={
+                                TeamType.SELF_PLAY_MEDIUM: 0.3, 
+                                TeamType.SELF_PLAY_HIGH: 0.3,
+                                TeamType.SELF_PLAY_ADVERSARY: 0.4,
+                            },
+                            probabilities_decay_over_time=0)
+    get_N_X_SP_agents(
+        args,
+        n_x_sp_train_types=curriculum.train_types,
+        n_x_sp_eval_types=primary_eval_types,
+        curriculum=curriculum,
+        unseen_teammates_len=unseen_teammates_len,
+        adversary_play_config=adversary_play_config,
+        attack_rounds=attack_rounds
+    )
+
+
+
+def N_1_SP(args) -> None:
     '''
     The randomly initialized agent will train with itself and one other unseen teammate (e.g. [SP, SP, SP, SP_H] in a 4-chef layout)
     
     :param pop_force_training: Boolean that, if true, indicates population should be generated, otherwise load it from file
     :param primary_force_training: Boolean that, if true, indicates the SP agent teammates_collection should be trained  instead of loaded from file
-    :param parallel: Boolean indicating if parallel envs should be used for training or not
+    :param  : Boolean indicating if   envs should be used for training or not
```

Comment:
We can delete these `:param` comments since the function signature no longer includes them